docs/data/data-grid/prompt/PromptWithDataSource.js

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  Unstable_GridToolbarPromptControl as GridToolbarPromptControl,
+  GridToolbar,
+} from '@mui/x-data-grid-premium';
+import { mockPromptResolver, useMockServer } from '@mui/x-data-grid-generator';
+import Stack from '@mui/material/Stack';
+import Box from '@mui/material/Box';
+
+function ToolbarWithPromptInput() {
+  return (
+    <Stack flex={1} gap={0.5} sx={{ px: 0.5 }}>
```

Comment:
```suggestion
    <Stack gap={0.5} sx={{ px: 0.5 }}>
```

To fix the gap when there are no rows. Same for other demos.
<img width="826" src="https://github.com/user-attachments/assets/2315a210-a78e-4eb3-b446-468888898c0e">